sbi/inference/snpe/snpe_base.py

```diff
@@ -299,9 +304,14 @@ def train(
 
             self.epoch += 1
 
+
+            log_prob_sum /= len(train_loader) * train_loader.batch_size
```

Comment:
I just realized: this will be inaccurate because we have the default `drop_last=False` for the `train_loader` (unlike for the validation loader, see [here](https://github.com/mackelab/sbi/blob/c9174e130cc622ae2219433dda541b74b37912a3/sbi/inference/base.py#L325-L335))

I think the best way to go is to estimate the number of training samples via the `validation_fraction`, i.e.
```
log_prob_sum /= int(tehta.shape[0] * validation_fraction)
```

Reply:
done with 1abbc88